### PR TITLE
fix screenshot upload

### PR DESCRIPTION
**Pull request by Exactor96**

### CODE REVIEW

File: src/lib/testrail.ts [hunk 1]
```diff
@@ -144,23 +146,19 @@ export class TestRail {
   }
 
   // This function will attach failed screenshot on each test result(comment) if founds it
-  public uploadScreenshots (caseId, resultId) {
-    const SCREENSHOTS_FOLDER_PATH = path.join(__dirname, 'cypress/screenshots');
+  public uploadScreenshots(caseId, resultId) {
+    const SCREENSHOTS_FOLDER_PATH = path.join(process.cwd(), 'cypress/screenshots');
```
Comment (Sc4ramouche) [inline src/lib/testrail.ts:150]: Question: what are the benefits of using `process.cwd` over `__dirname` here?

Comment (Sc4ramouche) [reply src/lib/testrail.ts:150]: Aha, I see, thanks! 

[StackOverflow: what's the difference between process.cwd() vs __dirname?](https://stackoverflow.com/questions/9874382/whats-the-difference-between-process-cwd-vs-dirname)

File: src/lib/utils.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            /* Recurse into a subdirectory */
+            results = results.concat(collectDirFiles(file));
+        } else {
+            /* Is a file */
+            results.push(file);
+        }
+    });
+    return results;
+}
+module.exports.collectDirFiles = collectDirFiles;
```
Comment (Sc4ramouche) [inline src/lib/utils.ts:21]: Question: this file appears to be a `.ts` file, can we add TypeScript annotation to this util function?

File: src/lib/testrail.ts [hunk 3]
```diff
@@ -144,23 +146,19 @@ export class TestRail {
...
-        if (file.includes(`C${caseId}`) && /(failed|attempt)/g.test(file)) {
-          try {
-            this.uploadAttachment(resultId, SCREENSHOTS_FOLDER_PATH + file)
-          } catch (err) {
-            console.log('Screenshot upload error: ', err)
-          }
+    files.forEach(file => {
+      if (file.includes(`C${caseId}`) && /(failed|attempt)/g.test(file)) {
+        try {
+          this.uploadAttachment(resultId, file)
```
Comment (Sc4ramouche) [inline src/lib/testrail.ts:157]: Question: we don't to prefix the filename with `SCREENSHOTS_FOLDER_PATH` anymore, correct? Just to double-check.

File: src/lib/testrail.ts [hunk 4]
```diff
@@ -6,6 +6,8 @@ const FormData = require('form-data');
 const TestRailLogger = require('./testrail.logger');
 const TestRailCache = require('./testrail.cache');
 import { TestRailOptions, TestRailResult } from './testrail.interface';
+const utils = require('./utils');
+const collectDirFiles = utils.collectDirFiles;
```
[the remark this answers is not in the document]
Comment (Exactor96) [reply src/lib/testrail.ts:10]: This is auto-import by IDE. Do you know the right way to import function?

File: src/lib/testrail.ts (hunk 3 above)
Comment (Exactor96) [reply src/lib/testrail.ts:157]: yeah, SCREENSHOTS_FOLDER_PATH is a part of the file here.

File: src/lib/testrail.ts [hunk 5]
```diff
@@ -44,23 +45,23 @@ export class TestRail {
...
-      .then(response => {
-        return response.data.map(item =>item.id)
+          username: this.options.username,
+          password: this.options.password
+        }
       })
-      .catch(error => console.error(error))
+        .then(response => {
+          return response.data.map(item => item.id)
+        })
```
Comment (Sc4ramouche) [inline src/lib/testrail.ts:58]: To keep it a bit more concise:

```suggestion
        .then(response => response.data.map(item => item.id))
```

File: src/lib/testrail.ts [hunk 6]
```diff
@@ -118,14 +119,14 @@ export class TestRail {
...
         data: JSON.stringify({ results }),
       })
-      .then(response => response.data)
-      .catch(error => { 
-        console.error(error); 
-      })
+        .then(response => response.data)
+        .catch(error => {
+          console.error(error);
+        })
```
Comment (Sc4ramouche) [inline src/lib/testrail.ts:125]: Could also save a line here :) 

```suggestion
        .catch(error => console.error(error))
```

File: src/lib/utils.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        if (stat && stat.isDirectory()) {
+            /* Recurse into a subdirectory */
+            results = results.concat(collectDirFiles(file));
+        } else {
+            /* Is a file */
+            results.push(file);
+        }
+    });
+    return results;
+}
```
Comment (Sc4ramouche) [inline src/lib/utils.ts:19]: What if we leverage a bit of ES6 here?

```suggestion
const collectDirFiles = (dir: string): string[] =>
  fsLib.readdirSync(dir).reduce((results: string[], fileName: string) => {
    const fullFilePath = path.join(dir, fileName);
    const stat = fsLib.statSync(file);

    if (stat && stat.isDirectory()) {
      return [...results, ...collectDirFiles(fullFilePath)];
    }

    return [...results, fullFilePath];
  }, []);

```

Didn't test it though, please let me know if this works.

**Review by Sc4ramouche — APPROVED** [review]

**Pull request merged**